src/main/java/com/facebook/openwifirrm/modules/ModelerUtils.java

```diff
@@ -202,4 +209,89 @@ public static double calculateTPCMetrics(
 			return Double.POSITIVE_INFINITY;
 		}
 	}
+
+	/**
+	 * Determines if two wifiscan entries should be aggregated without consideration
+	 * for any kind of obsoletion period. (Untested) should handle APs that support
+	 * pre-802.11n standards (no ht_oper and no vht_oper), 802.11n (supports ht_oper
+	 * but no vht_oper), and 802.11ac (supports both ht_oper and vht_oper).
+	 * Pre-802.11n, if two entries with the same bssid are in the same channel (and
+	 * therefore frequency, since they share a one-to-one mapping), that is enough
+	 * to aggregate them. However, 802.11n introduced channel bonding, so for
+	 * 802.11n onwards, channel width must be checked.
+	 *
+	 * @param entry1
+	 * @param entry2
+	 * @return true if the entries should be aggregated
+	 */
+	private static boolean matchesForAggregation(WifiScanEntry entry1, WifiScanEntry entry2) {
+		// TODO should we check the entire ht_oper and vht_oper or just channel width?
+		return entry1.bssid.equals(entry2.bssid) && Objects.equals(entry1.ssid, entry2.ssid)
```

Comment:
is there a reason that three different styles are employed here for equality comparison?

Reply:
For primitives, I do `==` (e.g., for an `int`). For Strings where it should never be null, I do `stringVariable.equals(otherObject)`. For Strings that could be `null` and where two null strings should count as being equal to each other, I do `Objects.equals`. Do you or @elludraon have a suggestion on a way to change this?

Reply:
@elludraon any thoughts on this?

Reply:
I would also use `Objects.equals` for `bssid` but otherwise this is fine.